### DISCUSSION TITLE
zhaquirks/xiaomi/aqara /motion_aq2b.py and motion_aq2.py

### DISCUSSION THREAD

**Discussion by vermaete** [zigpy/zha-device-handlers#2499] [Q&A]
There are two files in the directory zhaquirks/xiaomi/aqara.
Being motion_aq2.py and aq2b.py.

The class name in the two files is the same (MotionAQ2)
https://github.com/zigpy/zha-device-handlers/blob/dev/zhaquirks/xiaomi/aqara/motion_aq2.py#L34
https://github.com/zigpy/zha-device-handlers/blob/dev/zhaquirks/xiaomi/aqara/motion_aq2b.py#L30

The MODELS_INFO for the two is that same.

I'm still new in this repo, but wondering if the same class naming is not an error?

Br

**Comment by TheJulianJES** [comment]
If you look at the `ENDPOINTS` part of the device `signature` (and `replacement`), you can see that there are three cluster missing for the non-b variant:
```python
IasZone.cluster_id,	
PowerConfiguration.cluster_id,	
Identify.cluster_id,
```

The duplicate class name isn't an issue, because it's Python and the files/modules are called differently.
However, I do agree this is quote confusing and not optimal at all.
You could probably put both classes into one file/module (and rename one class) and then have one extend the other, so the `__init__` function can be de-duplicated.

So the two files/modules are for different sensors (which have the same model ID). I'd guess one of them is running a newer firmware version.
